apps/coursition/src/components/googleSignIn.tsx

```diff
@@ -0,0 +1,21 @@
+import { Button } from '@nmit-coursition/design-system'
+import { Loader } from 'lucide-react'
+import { signIn } from 'next-auth/react'
+import * as React from 'react'
+
+export const GoogleSignIn = () => {
+  const [isLoading, setIsLoading] = React.useState(false)
+
+  return (
+    <Button
+      onClick={async () => {
+        setIsLoading(true)
+        await signIn('google', { callbackUrl: '/' })
+      }}
+      className='w-ful text-white font-medium py-2 px-4 rounded-md'
+    >
+      <span>Sign in with google</span>
+      {isLoading && <Loader className='animate-spin text-sm ml-1' />}
+    </Button>
+  )
```

Comment:
**nitpick:** We should use server login (that works even without JS. But we'll handle it with Auth.js v5.
https://authjs.dev/getting-started/session-management/login

```suggestion
  return (
    <form
      action={async () => {
        "use server"
        await signIn("google", { callbackUrl: "/" })
      }}
    >
      <button type="submit">Sign in</button>
    </form>
  )
```